### PR TITLE
[MM-21921] Adds the server version check before running a command

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
Before running any command that uses the Client instance, `mmctl` will
now check that the server version matches its own, and will show an
error if it doesn't.

This PR enables as well the `--strict` flag and the `MMCTL_STRICT`
environment variable; if one of those is set, `mmctl` will error in case
the server version doesn't match and will not run the command.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-21921

### CODE REVIEW

File: commands/init.go [hunk 1]
```diff
@@ -4,58 +4,80 @@
...
 	return func(cmd *cobra.Command, args []string) error {
-		c, err := InitClient()
+		c, serverVersion, err := InitClient()
 		if err != nil {
 			return err
 		}
+
+		valid := CheckVersionMatch(Version, serverVersion)
+		if !valid {
+			if viper.GetBool("strict") {
```
Comment (ethervoid) [inline commands/init.go]: One question to understand better. Do we need viper as a new dependency?

I mean, we can just create a new method in `version.go` called something like `SetStrict` like we have with the `printer.SetFormat` and then use it here as we used `Version` variable in the line 32

File: Makefile [hunk 2]
```diff
@@ -7,7 +7,6 @@ BUILD_VERSION ?= $(shell git ls-remote --tags --refs --sort="v:refname" git://gi
 ADVANCED_VET ?= TRUE
 
 LDFLAGS += -X "github.com/mattermost/mmctl/commands.BuildHash=$(BUILD_HASH)"
-LDFLAGS += -X "github.com/mattermost/mmctl/commands.Version=$(BUILD_VERSION)"
```
Comment (ethervoid) [inline Makefile]: Just a comment, not a proposal. One thing that worries my about having version variables is that they tend to keep outdated.

File: commands/init_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		{
+			Name: "Minor version is less",
+			Version: "1.2.3",
+			ServerVersion: "1.1.3",
+			Expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
```
Comment (ethervoid) [inline commands/init_test.go]: :heart:

File: commands/init.go (hunk 1 above)
Comment (mgdelacroix) [reply commands/init.go]: I'm not sure I'm understanding your point here. Viper is used so when reading the `strict` value, it takes into consideration both the `--strict` flag and the `MMCTL_STRICT` environment variable. As a side benefit, I've binded as well the `MMCTL_FORMAT` environment variable, so now both can be received as flags or env variables.

The rationale behind this is that we want to allow the admins using `mmctl` that want all their commands to be strict to be able to do it easily, and an environment variable is a great mechanism for this.

Comment (ethervoid) [reply commands/init.go]: Cool, understood! Thank you

File: Makefile (hunk 2 above)
Comment (mgdelacroix) [reply Makefile]: I think that the version variable allows us to keep a different version number for each branch (say, if we have to release a 5.20 and a 5.19.1 versions of the tool), but it is true that we have to be careful for it not to become outdated.

**Pull request merged**